### DISCUSSION TITLE
Piping stream to s3 for ingest via HLS/dash manifest?

### DISCUSSION THREAD

**Discussion by lovgrandma** [bluenviron/mediamtx#1943] [Questions]
Hello I am building a service to allow livestreaming from OBS studio ingest and Im intending to pipe incoming stream transcoded into h.264 to s3 bucket in 720p and 360p.

I am using shaka packager, but my main questions are:

How to pipe incoming stream to S3?
Are streams from different clients handled individually? 

E.g user streaming from computer 1 is instanced differently from computer 2? I just need to set streamid and bucket on a struct separately in each instance to send A video from B user to C bucket with D stream ID.

**Comment by aler9** [comment]
Hello,

Regarding stream identification, in the server streams are splitted by path, not by user. The path name is a unique identifier of a stream.

Regarding HLS packaging, keep in mind that nowadays serving a static folder through HTTP is not enough to generate a Low-latency HLS stream, since the latest specification introduced preloading paths, that are paths that keep clients on hold until a certain part or segment is available, and this feature can't achieved by serving a static folder. Therefore, my advice is to generate HLS by using MediaMTX instances in the cloud and use Amazon CloudFront in front of them to distribute the load, instead of using S3 buckets. MediaMTX provides a HTTP/HLS server that supports preloading paths and all LL-HLS features.

[a comment by an automated account is not shown]
